src/daft-scan/src/lib.rs

```diff
@@ -536,7 +538,7 @@ impl Display for PartitionTransform {
     }
 }
 
-pub trait ScanOperator: Send + Sync + Debug {
+pub trait ScanOperator: Send + Sync + Debug + Any {
```

Comment:
Why is `Any` required?

Reply:
Ah good catch, that was an artifact of an attempt to add optional non-pointer-based equality to scan operators via downcasting! I'll remove that.